platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/chatter/ChatterGossip.java

```diff
@@ -0,0 +1,448 @@
+            @NonNull final AtomicReference<Consensus> consensusRef,
+            @NonNull final QueueThread<EventIntakeTask> intakeQueue,
+            @NonNull final FreezeManager freezeManager,
+            @NonNull final StartUpEventFrozenManager startUpEventFrozenManager,
+            @NonNull final SwirldStateManager swirldStateManager,
+            final boolean startedFromGenesis,
+            @NonNull final StateManagementComponent stateManagementComponent,
+            @NonNull final InterruptableConsumer<EventIntakeTask> eventIntakeLambda,
+            @NonNull final EventObserverDispatcher eventObserverDispatcher,
+            @NonNull final EventMapper eventMapper,
+            @NonNull final EventIntakeMetrics eventIntakeMetrics,
+            @NonNull final EventLinker eventLinker,
+            @NonNull final Runnable updatePlatformStatus,
+            @NonNull final Consumer<SignedState> loadReconnectState) {
+        super(
+                platformContext,
+                threadManager,
+                time,
+                crypto,
+                notificationEngine,
+                addressBook,
+                selfId,
+                appVersion,
+                shadowGraph,
+                consensusRef,
+                intakeQueue,
+                freezeManager,
+                startUpEventFrozenManager,
+                swirldStateManager,
+                stateManagementComponent,
+                eventMapper,
+                eventIntakeMetrics,
+                eventObserverDispatcher,
+                updatePlatformStatus,
+                loadReconnectState);
+
+        final BasicConfig basicConfig = platformContext.getConfiguration().getConfigData(BasicConfig.class);
+        final ChatterConfig chatterConfig = platformContext.getConfiguration().getConfigData(ChatterConfig.class);
+
+        chatterCore = new ChatterCore<>(
+                time,
+                GossipEvent.class,
+                new PrepareChatterEvent(CryptographyHolder.get()),
+                chatterConfig,
+                networkMetrics::recordPingTime,
+                platformContext.getMetrics());
+
+        reconnectController = new ReconnectController(threadManager, reconnectHelper, this::resume);
+
+        // first create all instances because of thread safety
+        for (final NodeId otherId : topology.getNeighbors()) {
+            chatterCore.newPeerInstance(otherId.id(), eventTaskCreator::addEvent);
+        }
+
+        if (emergencyRecoveryManager.isEmergencyStateRequired()) {
+            // If we still need an emergency recovery state, we need it via emergency reconnect.
+            // Start the helper first so that it is ready to receive a connection to perform reconnect with when the
+            // protocol is initiated.
+            thingsToStart.add(0, reconnectController::start);
+        }
+
+        intakeCycle = new SequenceCycle<>(eventIntakeLambda);
```

Comment:
This doesn't need to be done in this PR. But it seems like the `SequenceCycle` functionality should be added to `QueueThread` directly. The ability to wait for the current item to finish processing is something I could imagine other code needing as well, especially as we move toward the component model @lpetrovic05 is working on. If we did that, the intake queue would be passed directly into chatter instead of obscuring it via a lambda, then wrapping it and re-exposing it in the interface method.

Reply:
I agree, the API needed to support this is horrible.

Reply:
if we go with components, we would not pass around queue threads. it would have to be a task processing component feature. we will have to add this feature once we get to the intake component